Tests/MongoSwiftTests/ReadWriteConcernTests.swift

```diff
@@ -0,0 +1,213 @@
+    	defer { do { try db2.drop() } catch {} }
+
+    	// expect that a collection created from a DB with local RC also has local RC
+    	var coll3 = try db2.createCollection("coll3")
+    	expect(coll3.readConcern.level).to(equal("local"))
+
+    	// expect that a collection retrieved from a DB with local RC also has local RC
+    	coll3 = try db2.collection("coll3")
+    	expect(coll3.readConcern.level).to(equal("local"))
+
+    	// expect that a collection created from a DB with local RC can override the DB's RC
+    	var coll4 = try db2.createCollection("coll4", options: CreateCollectionOptions(readConcern: try ReadConcern(.majority)))
+    	expect(coll4.readConcern.level).to(equal("majority"))
+
+ 		// expect that a collection retrieved from a DB with local RC can override the DB's RC
+ 		coll4 = try db2.collection("coll4", options: CollectionOptions(readConcern: try ReadConcern(.majority)))
+    	expect(coll4.readConcern.level).to(equal("majority"))
+    }
+
+    func testOperationReadConcerns() throws {
+    	// setup a collection 
+    	let client = try MongoClient()
+    	let db = try client.db("test")
+    	defer { do { try db.drop() } catch {} }
+    	let coll = try db.createCollection("coll1")
+
+    	let command: Document = ["count": "coll1"]
+
+    	// run command with a valid readConcern
+    	let options1 = RunCommandOptions(readConcern: try ReadConcern(.local))
+    	let res1 = try db.runCommand(command, options: options1)
+        expect(res1["ok"] as? Double).to(equal(1.0))
+
+        // run command with an empty readConcern
+        let options2 = RunCommandOptions(readConcern: ReadConcern())
+        let res2 = try db.runCommand(command, options: options2)
+        expect(res2["ok"] as? Double).to(equal(1.0))
+
+        // running command with an invalid RC level should throw
+        let options3 = RunCommandOptions(readConcern: try ReadConcern("blah"))
+        expect(try db.runCommand(command, options: options3)).to(throwError())
+
+        // try various command + read concern pairs to make sure they work
+        expect(try coll.find(options: FindOptions(readConcern: try ReadConcern(.local)))).toNot(throwError())
+
+        expect(try coll.aggregate([["$project": ["a": 1] as Document]],
+        	options: AggregateOptions(readConcern: try ReadConcern(.majority)))).toNot(throwError())
+
+        expect(try coll.count(options: CountOptions(readConcern: try ReadConcern(.majority)))).toNot(throwError())
+
+        expect(try coll.distinct(fieldName: "a",
+        	options: DistinctOptions(readConcern: try ReadConcern(.local)))).toNot(throwError())
+    }
+
+    func testConnectionStrings() throws {
+    	let csPath = "\(self.getSpecsPath())/read-write-concern/tests/connection-string"
+    	let testFiles = try FileManager.default.contentsOfDirectory(atPath: csPath).filter { $0.hasSuffix(".json") }
+    	for filename in testFiles {
+    		let testFilePath = URL(fileURLWithPath: "\(csPath)/\(filename)")
```

Comment:
more weird space, there are a few below as well so I'll stop. Maybe turn on auto formatting in sublime?

Reply:
ugh I think this happened because I copy pasted some code from the `WriteConcern` branch on GitHub into this file - I'd realized a cleaner way to do the tests as I finished filling them out there but I guess that introduced a mix of tabs and spaces. anyway, yes, found a sublime plugin to autoconvert tabs to spaces on save for the future :)